tests/Object.ts

```diff
@@ -1057,10 +1057,36 @@ type O_PATHS = {
     c: boolean
 };
 
+type O_PATHS_DEEP = {
+    a: {
+        b: {
+            c: Date;
+            d: {
+                e: number;
+            };
+            g: {
+                f: number;
+            };
+            h: {r: number}[],
+        };
+    };
+};
+
 checks([
+    // Default compact mode
     check<O.Paths<{'prop': {a: 1}[]}>, T.NonNullable<['prop'?, number?, 'a'?]>, Test.Pass>(),
     check<O.Paths<O_PATHS>, T.NonNullable<['a'?, 'a'?] | ['b'?, 'a'?, 'a'?] | ['b'?, 'b'?] | ['c'?]>, Test.Pass>(),
-    check<O.Paths<O[]>, O.Paths<O[]>, Test.Pass>(),
```

Comment:
I removed this as it fails after my changes. To make it work you need to change the magic number 12 to 8... Since this happens due to circular ref I don't think we should support it. @millsp thoughts?